File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/proc/AttCastProcessor.java
package ru.citeck.ecos.records3.record.op.atts.proc;

import org.jetbrains.annotations.NotNull;
import ru.citeck.ecos.commons.data.DataValue;
import ru.citeck.ecos.commons.data.ObjectData;

import java.util.List;

public class AttCastProcessor extends AbstractAttProcessor<List<DataValue>> {

    @Override
    protected Object processOne(@NotNull ObjectData meta,
                                @NotNull DataValue value,
                                @NotNull List<DataValue> arguments) {

        if (arguments.size() == 0) {
            return value;
        }

        switch (arguments.get(0).asText()) {
            case "str":
                return value.asText();
            case "num":
                return value.asDouble();
            case "bool":
                return value.asBoolean();
        }

        return null;
    }

    @Override
    protected List<DataValue> parseArgs(@NotNull List<DataValue> arguments) {
        return arguments;
    }

    @NotNull
    @Override
    public String getType() {
        return "cast";
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/proc/AttCastProcessor.kt
package ru.citeck.ecos.records3.record.op.atts.service.proc

import ru.citeck.ecos.commons.data.DataValue
import ru.citeck.ecos.commons.data.ObjectData

class AttCastProcessor : AbstractAttProcessor<List<DataValue>>() {

    override fun processOne(attributes: ObjectData, value: DataValue, args: List<DataValue>): Any? {

        if (args.isEmpty()) {
            return value
        }
        when (args[0].asText()) {
            "str" -> return value.asText()
            "num" -> return value.asDouble()
            "bool" -> return value.asBoolean()
        }
        return null
    }

    override fun parseArgs(args: List<DataValue>): List<DataValue> {
        return args
    }

    override fun getType(): String = "cast"
}
